lending/loan_management/doctype/loan_repayment/loan_repayment.py

```diff
@@ -3007,79 +3007,103 @@ def post_bulk_payments(data):
 			", ".join(non_existent_loans)
 		)
 
-	grouped_by_loan = group_by_loan(data)
+	grouped_by_loan_and_loan_disbursement = group_by_loan_and_loan_disbursement(data)
 	# custom hash best
 	trace_id = random_string(10)
 
 	if frappe.flags.in_test:
-		bulk_repost(grouped_by_loan, trace_id)
+		bulk_repost(grouped_by_loan_and_loan_disbursement, trace_id)
 	else:
-		job = frappe.enqueue(bulk_repost, grouped_by_loan=grouped_by_loan, trace_id=trace_id)
+		job = frappe.enqueue(
+			bulk_repost, grouped_by_loan=grouped_by_loan_and_loan_disbursement, trace_id=trace_id
+		)
 		return {"job_id": job.id, "trace_id": trace_id}
 
 
-def group_by_loan(data):
-	grouped_by_loan = {}
+# grouping by disbursement because LoC loans exist
+# and it is easier to rollback per disbursement
+# than to rollback per loan
+def group_by_loan_and_loan_disbursement(data):
+	grouped_by_loan_and_dibsursement = {}
 	for repayment in data:
 		loan = repayment["against_loan"]
-		grouped_by_loan.setdefault(loan, [])
-		grouped_by_loan[loan].append(repayment)
-	return grouped_by_loan
+		disbursement = None
```

Comment:
There better ways to handle deterministic keys
You can define a key as `key = (repayment.loan, cstr(repayment.loan_disbursement))`

This seems like a better approach rather than searching for `loan_disbursement` key